flake.nix

```
@@ -47,8 +47,16 @@
                       })
                     ];
                   })).config;
+
+                moduleDeclarationFile =
+                  (builtins.unsafeGetAttrPos "modules" args).file;
               in
-              modules ++ [
+              (map (m:
+                if lib.isFunction m || lib.isAttrs m then
+                  { _file = moduleDeclarationFile; imports = [ m ]; }
+                else
+                  m)
```

I think we don't need the `if` here
```suggestion
                {
                  _file = moduleDeclarationFile;
                  imports = [ m ];
                }
```

---

You're right. I was a bit unsure whether this would eat up even more resources, but I guess it's not a big deal :)